### PR TITLE
Add child features aggregation function in select by value tool

### LINKED ISSUE #2230: [FEATURE] Search widget for relations using aggregates
Original commit: https://github.com/qgis/QGIS/commit/fdd00870eee1d8f06a8cedf50c3e5d5ab3611319 by m-kuhn

For each child relations, the subform is visible.

Each attribute of the children has a tool button option to define to which
aggregate the specified value should be compared. This allows for searching
things like

 * Each city where the highest building is more than 300 m
 * Each sensor where the median value is lower than 50 ppm
 * Each feature with a child with a missing value
 * ...

### PR DESCRIPTION
**Pull request by ghtmtt**
Goal: add 1:N relation selection option to `Select By Value` tool

Ticket(s): fix #2230

- [x] Backport to LTR documentation is required

### CODE REVIEW

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 1]
```diff
@@ -810,6 +810,18 @@ Here it is possible to digitize a polygon for the region layer using the |signPl
 if you chose the option ``Allow adding new features`` in the 
 :menuselection:`Properties --> Attributes Form` menu of the airport layer.
 
+Child features can be used also with the :ref:`select_by_value` tool in
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/attribute_table.rst:813]:
```suggestion
Child layer can be used also with the :ref:`select_by_value` tool in
```
?

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 2]
```diff
@@ -810,6 +810,18 @@ Here it is possible to digitize a polygon for the region layer using the |signPl
...
+order to select features of the parent layer depending on children attributes.
+
+You might want to select all the regions where the *mean* altitude of the
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/attribute_table.rst:816]: It took me some time to "understand" why people might want to do that. Maybe directly state that 
```suggestion
As an example, the :ref:`figure_select_by_value` selects all the regions where the altitude of the
```

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 3]
```diff
@@ -810,6 +810,18 @@ Here it is possible to digitize a polygon for the region layer using the |signPl
 if you chose the option ``Allow adding new features`` in the 
 :menuselection:`Properties --> Attributes Form` menu of the airport layer.
 
+Child features can be used also with the :ref:`select_by_value` tool in
+order to select features of the parent layer depending on children attributes.
+
+You might want to select all the regions where the *mean* altitude of the 
+airports is greater than 500 meters above sea level. 
+
+Many different aggregation functions are available in the form:
```
Comment (DelazJ) [inline docs/user_manual/working_with_vector/attribute_table.rst:819]: These are not aggregation functions, are they?

Comment (ghtmtt) [reply docs/user_manual/working_with_vector/attribute_table.rst:819]: actually yes. In the example you can have more airports in the same region. So you select the region where the average altitude of all the airports is greater than 500.

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 4]
```diff
@@ -810,6 +810,18 @@ Here it is possible to digitize a polygon for the region layer using the |signPl
...
+
+.. _figure_select_by_value:
+
+.. figure:: img/relation_select_by_value.png
+   :align: center
```
[the remark this answers is not in the document]
Comment (ghtmtt) [reply docs/user_manual/working_with_vector/attribute_table.rst:824]: done

File: docs/user_manual/working_with_vector/attribute_table.rst (hunk 3 above)
Comment (DelazJ) [reply docs/user_manual/working_with_vector/attribute_table.rst:819]: Ah OK, I did not notice that there are two levels of buttons for children

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 5]
```diff
@@ -810,6 +810,20 @@ Here it is possible to digitize a polygon for the region layer using the |signPl
 if you chose the option ``Allow adding new features`` in the 
 :menuselection:`Properties --> Attributes Form` menu of the airport layer.
 
+Child layer can be used also with the :ref:`select_by_value` tool in
+order to select features of the parent layer depending on children attributes.
```
Comment (havatv) [inline docs/user_manual/working_with_vector/attribute_table.rst:814]:
```suggestion
The child layer can also be used in the :ref:`select_by_value` tool in
order to select features of the parent layer based on attributes of their children.
```

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 6]
```diff
@@ -810,6 +810,20 @@ Here it is possible to digitize a polygon for the region layer using the |signPl
...
+
+As an example, the :ref:`figure_select_by_value` selects all the regions where 
+the altitude of the airports is greater than 500 meters above sea level.
```
Comment (havatv) [inline docs/user_manual/working_with_vector/attribute_table.rst:817]:
```suggestion
In :ref:`figure_select_by_value`, all the regions where the mean altitude of
the airports is greater than 500 meters above sea level are selected. 
```

File: docs/user_manual/working_with_vector/attribute_table.rst [hunk 7]
```diff
@@ -810,6 +810,20 @@ Here it is possible to digitize a polygon for the region layer using the |signPl
 if you chose the option ``Allow adding new features`` in the 
 :menuselection:`Properties --> Attributes Form` menu of the airport layer.
 
+Child layer can be used also with the :ref:`select_by_value` tool in
+order to select features of the parent layer depending on children attributes.
+
+As an example, the :ref:`figure_select_by_value` selects all the regions where 
+the altitude of the airports is greater than 500 meters above sea level. 
+
+Many different aggregation functions are available in the form:
```
Comment (havatv) [inline docs/user_manual/working_with_vector/attribute_table.rst:819]:
```suggestion
You will find that many different aggregation functions are available in the form.
```

**Pull request merged**